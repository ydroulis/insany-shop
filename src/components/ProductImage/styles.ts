"use client"

import Image from 'next/image'
import styled from 'styled-components'

export const Container = styled(Image)`
    border-radius: 4px;

    width: 100%;
    height: 100%;
    @media (min-width: 1440px) {
        width: 640px;
        min-width: 640px;
        height: 580px;
    }
`