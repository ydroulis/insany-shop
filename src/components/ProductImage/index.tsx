import React from 'react';

import * as S from './styles';

interface ProductImageProps {
    image: string
    name: string
}

const ProductImage: React.FC<ProductImageProps> = ({ image, name }) => {
    return (
        <S.Container
            src={image}
            alt="Imagem do produto"
            width={640}
            height={580}
            loading="lazy"
            placeholder='blur'
            blurDataURL="..."
            role="img"
            aria-label={`Imagem ilustrativa do produto ${name}`}
            sizes="(max-width: 768px) 100vw, 640px"
        />
    );
}

export default ProductImage;