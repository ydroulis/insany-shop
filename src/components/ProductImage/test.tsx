import React from "react";
import { render, screen } from "@testing-library/react";
import ProductImage from "./";

describe("<ProductImage />", () => {
    const defaultProps = {
        image: "/fake-product.jpg",
        name: "Tênis Esportivo",
    };

    it("should render image with required attributes", () => {
        render(<ProductImage image="/fake-product.jpg" name="Tênis Esportivo" />);

        const img = screen.getByAltText('Imagem do produto');

        expect(img).toBeInTheDocument();
        expect(img).toHaveAttribute("src");
        expect(img).toHaveAttribute("alt");
    });

    it("should include aria-label with the product name", () => {
        render(<ProductImage {...defaultProps} />);
        const img = screen.getByRole("img");

        expect(img).toHaveAttribute(
            "aria-label",
            `Imagem ilustrativa do produto ${defaultProps.name}`
        );
    });

    it("should apply width and height", () => {
        render(<ProductImage {...defaultProps} />);
        const img = screen.getByRole("img");

        expect(img).toHaveAttribute("width", "640");
        expect(img).toHaveAttribute("height", "580");
    });

    it("should match snapshot", () => {
        const { container } = render(<ProductImage {...defaultProps} />);
        expect(container).toMatchSnapshot();
    });
});
