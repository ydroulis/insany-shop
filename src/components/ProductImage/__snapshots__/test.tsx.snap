// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<ProductImage /> should match snapshot 1`] = `
<div>
  <img
    alt="Imagem do produto"
    aria-label="Imagem ilustrativa do produto TÃªnis Esportivo"
    class="sc-bRKDuR ipYYMt"
    data-nimg="1"
    decoding="async"
    height="580"
    loading="lazy"
    role="img"
    sizes="(max-width: 768px) 100vw, 640px"
    src="/_next/image?url=%2Ffake-product.jpg&w=3840&q=75"
    srcset="/_next/image?url=%2Ffake-product.jpg&w=640&q=75 640w, /_next/image?url=%2Ffake-product.jpg&w=750&q=75 750w, /_next/image?url=%2Ffake-product.jpg&w=828&q=75 828w, /_next/image?url=%2Ffake-product.jpg&w=1080&q=75 1080w, /_next/image?url=%2Ffake-product.jpg&w=1200&q=75 1200w, /_next/image?url=%2Ffake-product.jpg&w=1920&q=75 1920w, /_next/image?url=%2Ffake-product.jpg&w=2048&q=75 2048w, /_next/image?url=%2Ffake-product.jpg&w=3840&q=75 3840w"
    style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
    width="640"
  />
</div>
`;
